From 221dff2e2f9fbd1c18eae9b59f23b1dbbdb1870b Mon Sep 17 00:00:00 2001
From: Aleix Pol <aleixpol@kde.org>
Date: Mon, 5 May 2014 15:13:25 +0200
Subject: [PATCH 1/2] Fix crash

If we don't have a primary output, don't query it. Set a new primary
output instead.

Reviewed by Alex Fiestas
---
 backends/xrandr/xrandrconfig.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backends/xrandr/xrandrconfig.cpp b/backends/xrandr/xrandrconfig.cpp
index 0236a3a..a842173 100644
--- a/backends/xrandr/xrandrconfig.cpp
+++ b/backends/xrandr/xrandrconfig.cpp
@@ -157,7 +157,7 @@ void XRandRConfig::updateKScreenConfig(Config *config) const
         output->updateKScreenOutput(kscreenOutput);
     }
 
-    if (config->primaryOutput()->id() != m_primaryOutput) {
+    if (!config->primaryOutput() || config->primaryOutput()->id() != m_primaryOutput) {
         config->setPrimaryOutput(config->output(m_primaryOutput));
     }
 }
-- 
2.0.0.rc2

